pkg/riff/commands/credential_set.go

```diff
@@ -49,42 +97,198 @@ func NewCredentialSetCommand(c *cli.Config) *cobra.Command {
 		Args: cli.Args(
 			cli.NameArg(&opts.Name),
 		),
-		PreRunE: cli.ValidateOptions(opts),
-		RunE: func(cmd *cobra.Command, args []string) error {
-			secret, err := c.Core().Secrets(opts.Namespace).Get(opts.Name, metav1.GetOptions{})
-			if err != nil {
-				if !apierrs.IsNotFound(err) {
-					return err
+		PreRunE: func(cmd *cobra.Command, args []string) error {
+			if opts.DockerHubId != "" || opts.RegistryUser != "" {
+				// capture option from stdin
+
+				var prompt string
+				var value *string
+
+				if opts.DockerHubId != "" {
+					prompt = "Docker Hub password"
+					value = &opts.DockerHubPassword
+				}
+				if opts.RegistryUser != "" {
+					prompt = "Registry password"
+					value = &opts.RegistryPassword
 				}
 
-				_, err = c.Core().Secrets(opts.Namespace).Create(&corev1.Secret{
-					ObjectMeta: metav1.ObjectMeta{
-						Name:      opts.Name,
-						Namespace: opts.Namespace,
-						Labels: map[string]string{
-							// TODO get label from riff system
-							"projectriff.io/credential": "",
-						},
-					},
-					// TODO define secret data
-					StringData: map[string]string{},
-				})
+				if terminal.IsTerminal(int(syscall.Stdin)) {
+					fmt.Printf("%s: ", prompt)
+					res, err := terminal.ReadPassword(int(syscall.Stdin))
+					fmt.Println("")
+					if err != nil {
+						return err
+					}
+					*value = string(res)
+				} else {
+					reader := bufio.NewReader(c.Stdin)
+					res, err := ioutil.ReadAll(reader)
+					if err != nil {
+						return err
+					}
+					*value = string(res)
+				}
+			}
+
+			// continue with option validation
+			return cli.ValidateOptions(opts)(cmd, args)
+		},
+		RunE: func(cmd *cobra.Command, args []string) error {
+			// get desired credential and image prefix
+			secret, defaultImagePrefix, err := makeCredential(opts)
+			if err != nil {
 				return err
 			}
 
-			// ensure we are not mutating a non-riff secret
-			if _, ok := secret.Labels["projectriff.io/credential"]; !ok {
-				return fmt.Errorf("credential %q exists, but is not owned by riff", opts.Name)
+			if err := setCredential(c, opts, secret); err != nil {
+				return err
 			}
+			fmt.Fprintf(cmd.OutOrStdout(), "Set credentials %q\n", opts.Name)
 
-			// TODO mutate secret
-			_, err = c.Core().Secrets(opts.Namespace).Update(secret)
+			if opts.SetAsDefault {
+				if defaultImagePrefix == "" {
+					fmt.Fprintf(cmd.OutOrStdout(), "Unable to derive default image prefix\n")
+				} else {
+					err := setDefaultImagePrefix(c, opts, defaultImagePrefix)
+					if err != nil {
+						return err
+					}
+					fmt.Fprintf(cmd.OutOrStdout(), "Set default image prefix to %q\n", defaultImagePrefix)
+				}
+			}
 
-			return err
+			return nil
 		},
 	}
 
 	cli.NamespaceFlag(cmd, c, &opts.Namespace)
+	cmd.Flags().StringVar(&opts.DockerHubId, "docker-hub", "", "<todo>")
```

Comment:
Missing help text.

Reply:
Yes, All help text is missing. I've intentionally deferred writing help so that we can address it comprehensively and consistently for all commands. There are 99 `<todo>`s currently.